File: Python/A_Project/B_Snake_and_Food/Utility/text_board_renderer.py
from Utility.pair import Pair


class TextBoardRenderer:
    def render(self, board, snake_body, current_food):
        height = board.get_height()
        width = board.get_width()

        body_set = set(snake_body)
        head = snake_body[0] if len(snake_body) > 0 else None

        for r in range(height):
            row_cells = []
            for c in range(width):
                cell = "."
                if head is not None and head.row == r and head.col == c:
                    cell = "H"
                elif Pair(r, c) in body_set:
                    cell = "S"
                elif current_food is not None and current_food[0] == r and current_food[1] == c:
                    cell = "F"
                row_cells.append(cell)
            print(" ".join(row_cells))
        print()

File: Ruby/A_Project/B_Snake_And_Food/Utility/text_board_renderer.rb
# frozen_string_literal: true

require "set"

require_relative "pair"

class TextBoardRenderer
  def render(board, snake_body, current_food)
    height = board.get_height
    width = board.get_width
    body_set = snake_body.to_set
    head = snake_body[0]

    height.times do |row|
      row_cells = []
      width.times do |col|
        cell = "."
        if !head.nil? && head.row == row && head.col == col
          cell = "H"
        elsif body_set.include?(Pair.new(row, col))
          cell = "S"
        elsif !current_food.nil? && current_food[0] == row && current_food[1] == col
          cell = "F"
        end
        row_cells << cell
      end
      puts row_cells.join(" ")
    end
    puts
  end
end
